src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -83,7 +83,7 @@ public function getConfigTreeBuilder(): TreeBuilder
             ->end()
             ->fixXmlConfig('enabled_locale')
             ->children()
-                ->scalarNode('secret')->end()
+                ->scalarNode('secret')->defaultValue('')->end()
```

Comment:
This fixes a compiler error related to a missing `kernel.secret` parameter in the `uri_signer` service. This service will raise an exception if it is instantiated with an empty secret anyway.

Reply:
But what if a third-party service is using the empty secret and does not throw an exception in such a case?

Reply:
Yeah, same case for new projects. Note that the `APP_SECRET` environment var is now empty by default in the [recipe](https://github.com/symfony/recipes/pull/1317/files#diff-109059f345c5d69d28c125218f07442272ede7f8732541db37f8b4e4ebaa020dR16). This means that the application should work with this empty value. Here I wanted to apply the same principle but for situations where there is no config at all.

In my opinion, this topic is not perfect yet. I’d like services to complain about the secret value being empty at runtime without requiring them to do it on their own. I'm wondering if the `LazyString` could help here… any other idea?

Reply:
Using a factory that returns a string as a service could be a way to go.

Reply:
Reverted in favor of a lazy parameter resolution instead